src/Compilers/CSharp/Portable/Compilation/CSharpCompilerDiagnosticAnalyzer.cs

```diff
@@ -57,6 +57,7 @@ internal override ImmutableArray<int> GetSupportedErrorCodes()
                     case (int)ErrorCode.WRN_MainCantBeGeneric:
                     case (int)ErrorCode.ERR_NoMainInClass:
                     case (int)ErrorCode.ERR_MainClassNotFound:
+                    case (int)ErrorCode.WRN_SyncAndAsyncEntryPoints:
```

Comment:
Is there a test covering this change?

Reply:
@333fred I'm actually not sure what is it for. I added this because it seems like all entry point diagnostics are grouped there 😄

Reply:
@333fred [This](https://github.com/dotnet/roslyn/blob/1ed3ef5f8a7f4e4264430ab3787c9dd7a7a817c6/src/Compilers/CSharp/Test/Semantic/Diagnostics/DiagnosticAnalyzerTests.cs#L707) does seem to a related test for this. Without adding this line, that test would fail and report "Add resource string named '{0}' for Title of '{1}' to '{2}'" on the new warning.
I don't know why to exclude a warning from being tested on having a title?

Currently, `WRN_MainCantBeGeneric` **does** have a `_Title`, but if it's removed, the test won't fail.

Is the title for these specific diagnostics not used? In that case, `WRN_MainCantBeGeneric_Title` would be extra and unneeded.

Or if the title is used, then the test shouldn't exclude these.

Reply:
I think the `_Title` for warnings is used in the Visual Studio settings page that lists compiler warnings and allows you to disable them.

Reply:
Don't make any modifications to other warnings in this PR please.